### PR TITLE
UXENG-2937 - Update deps

### PR DESCRIPTION
**Pull request by bnlb**
Ken asked for an estimate on how long this would take to do, and in looking into it it turned out to be faster to just update than I had thought, so I just went ahead and updated everything.

⚠️ There was a major version bump of Jekyll from v3 to v4, but nothing appears to be effected and things work fine locally. I'll do more tests in test once this has been merged before we deploy anything, but if someone else could fire up localhost I'd appreciate it.

https://algorithmia.atlassian.net/browse/UXENG-2937

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ app.get('/metrics', (req, res, next) => {
   // empty bearer token.
   if (expectedToken && segments.length === 2 && segments[1] === expectedToken) {
     res.set('Content-Type', prometheus.register.contentType)
-    res.end(prometheus.register.metrics())
+    res.end(await prometheus.register.metrics())
```
Comment (bnlb) [inline server/index.js:81]: The changes happening here are identical to [what happened in](https://github.com/algorithmiaio/web/pull/2552) web recently.

File: package.json [hunk 2]
```diff
@@ -36,18 +36,19 @@
...
     "mocha": "^6.2.0",
     "shelljs": "^0.8.3"
   },
   "dependencies": {
-    "bunyan": "^1.8.12",
-    "cross-env": "^6.0.0",
+    "bunyan": "^1.8.15",
+    "cross-env": "^7.0.3",
     "express": "^4.17.1",
-    "http-proxy-middleware": "^0.20.0",
```
Comment (bnlb) [inline package.json:46]: This isn't used if NODE_ENV === 'production', so moved to dev dependencies without updating.

File: package.json [hunk 3]
```diff
@@ -9,21 +9,21 @@
...
+    "e2e:local": "E2E_BASE_URL=http://localhost:3000 yarn e2e",
+    "setup": "yarn apidocs:setup && yarn apidocs:build && yarn synapse:setup && gem install bundler && bundle install && yarn",
     "server": "node ./server/index.js",
-    "server:dev": "NODE_ENV=development npm run server",
-    "server:prod": "NODE_ENV=production npm run server",
-    "start": "npm run server:prod",
-    "synapse:setup": "git submodule init && git submodule update && cd synapse && npm ci",
+    "server:dev": "NODE_ENV=development yarn server",
+    "server:prod": "NODE_ENV=production yarn server",
+    "start": "yarn server:prod",
```
Comment (bnlb) [inline package.json:25]: Unrelated to this change but moved from npm to yarn so things are more consistent across repos.

File: Dockerfile [hunk 4]
```diff
@@ -72,9 +72,9 @@ COPY --from=docs-builder /opt/builds/build ./docs
 COPY server/index.js ./server/index.js
 COPY server/prometheus.js ./server/prometheus.js
 COPY config ./config
-COPY package.json package-lock.json ./
+COPY package.json yarn.lock ./
 
-RUN npm install --production
+RUN yarn --frozen-lockfile
```
Comment (warmbowski) [inline Dockerfile]: great!

File: Gemfile.lock [hunk 5]
```diff
@@ -32,39 +32,44 @@ GEM
...
-    jekyll-sass-converter (1.5.2)
-      sass (~> 3.4)
+    jekyll-sass-converter (2.1.0)
+      sassc (> 2.0.1, < 3.0)
     jekyll-sitemap (1.4.0)
       jekyll (>= 3.7, < 5.0)
     jekyll-watch (2.2.1)
       listen (~> 3.0)
-    kramdown (1.17.0)
+    kramdown (2.3.0)
```
Comment (warmbowski) [inline Gemfile.lock:63]: cool. looked like this package was the biggest security problem in the repo.

**Review by warmbowski — DISMISSED** [review]
LGTM

File: Dockerfile (hunk 4 above)
Comment (warmbowski) [reply Dockerfile]: Now that I look at it. do you not want to add --production anymore and suppress installing devDeps?

Comment (bnlb) [reply Dockerfile]: Good call, not sure why I didn't think to do that. Will update!

**Pull request merged**